TestUtilities/Sources/TestUtilities/Helpers/DDAssert.swift

```diff
@@ -0,0 +1,209 @@
+    }
+
+    if mirror1.children.isEmpty, mirror1.children.isEmpty {
+        guard String(describing: value1) == String(describing: value2) else { // plain values, compare debug strings
+            throw DDAssertError.expectedFailure("(\"\(value1)\") is not equal to (\"\(value2)\")", keyPath: keyPath)
+        }
+
+        return
+    }
+
+    switch (mirror1.displayStyle, mirror2.displayStyle) {
+    case (.dictionary?, .dictionary?): // two dictionaries
+        let dictionary1 = value1 as! [String: Any]
+        let dictionary2 = value2 as! [String: Any]
+
+        guard dictionary1.keys.count == dictionary2.keys.count else {
+            throw DDAssertError.expectedFailure("dictionaries have different number of keys", keyPath: keyPath)
+        }
+
+        for (key1, value1) in dictionary1 {
+            guard let value2 = dictionary2[key1] else {
+                throw DDAssertError.expectedFailure("dictionaries have different key names", keyPath: keyPath)
+            }
+
+            try _DDAssertReflect(value1, value2, keyPath: keyPath + [key1])
+        }
+
+        return // dictionaries are equal
+    case (.set?, .set?): // two sets
+        let set1 = value1 as! Set<AnyHashable>
+        let set2 = value2 as! Set<AnyHashable>
+
+        guard set1 == set2 else {
+            throw DDAssertError.expectedFailure("sets are not equal", keyPath: keyPath)
+        }
+
+        return // sets are equal
+    default:
+        break // other than dictionary or set, continue...
+    }
+
+    for (child1, child2) in zip(mirror1.children, mirror2.children) { // compare each child
+        let key = child1.label.map { [$0] } ?? []
+        try _DDAssertReflect(child1.value, child2.value, keyPath: keyPath + key)
+    }
+}
+
+public func DDAssertReflect<T>(_ expression1: @autoclosure () throws -> T, _ expression2: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #filePath, line: UInt = #line) {
+    _DDEvaluateAssertion(message: message(), file: file, line: line) {
+        try _DDAssertReflect(expression1(), expression2())
+    }
+}
+
+public func DDAssertNotReflect<T>(_ expression1: @autoclosure () throws -> T, _ expression2: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #filePath, line: UInt = #line) {
+    _DDEvaluateAssertion(message: message(), file: file, line: line) {
+        do {
+            try _DDAssertReflect(expression1(), expression2())
+        } catch DDAssertError.expectedFailure {
+            return // expected
+        }
+
+        throw DDAssertError.expectedFailure("Reflections are equal")
+    }
+}
+
+private func _DDAssertJSONEqual<T, U>(_ expression1: @autoclosure () throws -> T, _ expression2: @autoclosure () throws -> U, _ message: @autoclosure () -> String = "", file: StaticString = #filePath, line: UInt = #line) throws where T: Encodable, U: Encodable {
```

Comment:
These parameters aren't necessary as we evaluate failures in `_DDEvaluateAssertion` 🙌 
```suggestion
private func _DDAssertJSONEqual<T, U>(_ expression1: @autoclosure () throws -> T, _ expression2: @autoclosure () throws -> U) throws where T: Encodable, U: Encodable {
```